enhancements/update/change-management-and-maintenance-schedules.md

```diff
@@ -0,0 +1,1408 @@
+updated.
+  
+### API Extensions
+
+The enhancement is API driven, as described in the overview. Two new CRDs are introduced:
+- HostedChangeManagementPolicy
+- ChangeManagementPolicy
+
+Four existing APIs are modified in order to support change management policy definitions:
+- HostedCluster
+- NodePool
+- ClusterVersion
+- MachineConfigPool
+
+Support for change management policies requires behavioral changes in controllers associated with
+these resources to honor configured policies. 
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+In the HCP topology, the `HostedCluster` and `NodePool` resources are enhanced to support the `spec.changeManagement`
+stanza. 
+
+#### Standalone Clusters
+
+In the Standalone topology, the ClusterVersion and MachineConfigPool resources are enhanced to support the 
+`spec.changeManagement` stanza.
+
+#### Single-node Deployments or MicroShift
+
+These toplogies do not have worker nodes. Only the ClusterVersion change management policy will be relevant.
+There is no logical distinction between user workloads and control-plane workloads in this case. A control-plane
+update will drain user workloads and will cause workload disruption.
+
+Though disruption is inevitable, the maintenance schedule feature provides value in these toplogies by
+giving the cluster lifecycle administrator high level controls on exactly when it occurs.
+
+#### OCM Managed Profiles
+OpenShift Cluster Manager (OCM) should expose a user interface allowing users to manage their change management policy.
+Standard Fleet clusters will expose the option to configure the control-plane and worker-node `[Hosted]ChangeManagementPolicy`
+objects with the `MaintenanceSchedule` strategy - including permit and exclude times.
+
+- Service Delivery will reserve the right to override this policy for emergency corrective actions.
+- Service Delivery should constrain permit & exclude configurations based on their internal policies. 
+  For example, customers may be forced to enable permissive windows which amount to at least 6 hours a month.
+
+### Implementation Details/Notes/Constraints
+
+#### ChangeManagement Stanza
+The `spec.changeManagement` stanza will be introduced into ClusterVersion and MachineConfigPool (for standalone profiles)
+and HostedCluster and NodePool (for HCP profiles).
+
+#### MaintenanceSchedule Strategy Configuration
+
+When a `[Hosted]ChangeManagementPolicy` is defined to use the `MaintenanceSchedule` strategy,
+a `maintenanceSchedule` stanza should also be provided to configure the strategy
+(if it is not, the policy is functionally identical to `Restrictive`).
+
+```yaml
+kind: ChanageManagementPolicy
+spec:
+  strategy: MaintenanceSchedule
+ 
+  maintenanceSchedule:
+
+    # Specifies a reoccurring permissive window. 
+    permit:  
+      
+      # If null, all dates are permitted and only exclude constrains permissive windows.
+      recurrence: 

```

Comment:
Is there every likely to be a scenario where you cannot express the combination of dates you want, to a point where we might need multiple recurrences? (ie should this be a list?)

Scanning the API below, I'm struggling to think of one

Reply:
With sufficiently complex requirements, I would expect a customer to directly manage a ChangeManagementPolicy -- toggling it from permissive to restrictive. So there is an escape valve.